modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/negotiation/LeaseNegotiator.java

```diff
@@ -90,6 +90,8 @@ public void negotiate(LeaseAgreement agreement) {
                             LOG.warn("Lease was not negotiated due to exception [lease={}]", throwable, lease);
                         }
 
+                        leaseToNegotiate.remove(agreement.groupId());
```

Comment:
I am not sure that we can remove agreement by the group id. My concern is whether we are sure the agreement coincides with what is in leaseToNegotiate map.

Reply:
changed to `leaseToNegotiate.remove(agreement.groupId(), agreement);`